Initial

andrasguseo opened the pull request:
Initial commit

cliffordp commented on src/event-progress-bar.js:
```diff
@@ -0,0 +1,63 @@
+// Update the count down every 1 second
+var x = setInterval( doProgress, 1000 );
```
We don't want a very generically named var (`x`) in the global scope.

andrasguseo replied:
Done.

cliffordp commented on src/event-progress-bar.js:
```diff
@@ -0,0 +1,63 @@
+	var d = new Date();
+	var now = d.getTime();
+	var tzOffset = d.getTimezoneOffset();
+	var offset = tzOffset * 60 * 1000;
```
Not all time zones are hours. Some are 30min or 45min.
Why are we doing this at all? (e.g. `timeLeft = endDate - (now + offset);`)
Why not use [MomentJS](https://momentjs.com/), as provided by TEC?

andrasguseo replied:
We are getting the events start and end times in UTC time.

`now` gives you the current time in your own timezone.
The `offset` will give you the offset from UTC in seconds.
Later down we use `(now + offset)` to get the current UTC time.

This way everything (event times and the current time) is in UTC, so the calculations will be right.

FYI: I did not change anything in the code.

cliffordp commented on tribe-ext-event-progress-bar.php:
```diff
@@ -0,0 +1,268 @@
+		 */
+		public function progressbar( $file, $name, $template ) {
+
+			$event = tribe_get_event( get_the_ID() );
```
Doesn't `tribe_get_event()` get the global post by default?
```suggestion
			$event = tribe_get_event();
```

andrasguseo replied:
I guess it does. :) Tested, works.
Done.

cliffordp commented on tribe-ext-event-progress-bar.php:
```diff
@@ -0,0 +1,268 @@
+			/**
+			 * Leaving here for testing purposes.
+			 *
+			 * @todo Remove before publishing
```
TODO

andrasguseo replied:
Done.

cliffordp commented on tribe-ext-event-progress-bar.php:
```diff
@@ -0,0 +1,268 @@
+
+			// Bail if event (start) is not in the past, so event hasn't started yet
+			if ( ! $event->is_past ) {
+				/* ?>
+				<div>Event is coming.</div>
+				<?php */
```
Remove commented out code?

cliffordp replied:
Use TEC's `tribe_is_past_event()`?

cliffordp replied:
Consider how https://github.com/moderntribe/the-events-calendar/blob/5.2.1/src/Tribe/Template/Month.php#L1172-L1173 determines if an event is in the future - might be the same logic you need for "if has not yet started".

andrasguseo replied:
On `tribe_is_past_event()`:

`tribe_is_past_event()` works differently vs. `! $event->is_past`

`tribe_is_past_event()` is true when the event end date is in the past.
`$event->is_past` is true when the event start date is in the past.

In reference:
`tribe_is_past_event()` is false when the event has started (start date in past) but has not finished yet (end date in future). If we bail then, then the progress bar will not show up for the event that is currently live.

`$event->is_past` is false when the start date AND then end date are in the future.

As a consequence `! $event->is_past` will bail (not show the progress bar) for events whose start date is in the future (not in the past).

`! tribe_is_past_event()` will bail (not show the progress bar) for events that are currently live, where we actually need them to show.

cliffordp commented on tribe-ext-event-progress-bar.php:
```diff
@@ -0,0 +1,268 @@
+			?>
+
+			<div class="progress-bar-container progress-bar-container__on tribe-common-b2">
+				<div class="progress-bar-container__live" data-start-date="<?php echo $start_date ?>" data-end-date="<?php echo $end_date ?>">
```
`esc_attr()`

andrasguseo replied:
Oh good point. Learning.
Done.

cliffordp commented on tribe-ext-event-progress-bar.php:
```diff
@@ -0,0 +1,268 @@
+					<?php
+					printf(
+						// translators: %1$s: The remaining time with markup, %2$s: Closing </span>.
+						esc_html_x( '%1$s left.%2$s', 'The remaining time of a live event.', 'tribe-ext-event-progress-bar' ),
```
```suggestion
						esc_html_x( '%1$s left.%2$s', 'The remaining time of a live event, including HTML.', 'tribe-ext-event-progress-bar' ),
```

andrasguseo replied:
Done.

cliffordp commented on tribe-ext-event-progress-bar.php:
```diff
@@ -0,0 +1,268 @@
+	 */
+	class Main extends Tribe__Extension {
+
+		/**
+		 * Is Events Calendar PRO active. If yes, we will add some extra functionality.
+		 *
+		 * @return bool
+		 */
+		public $ecp_active = false;
+
+		/**
+		 * Is Event Tickets active. If yes, we will add some extra functionality.
+		 *
+		 * @return bool
+		 */
+		public $et_active = false;
+
```
```suggestion

```

andrasguseo replied:
Done

cliffordp commented on tribe-ext-event-progress-bar.php:
```diff
@@ -0,0 +1,268 @@
+		 * This always executes even if the required plugins are not present.
+		 */
+		public function construct() {
+			$this->add_required_plugin( 'Tribe__Events__Main', '5.0' );
```
Add comment why the version you required.

andrasguseo replied:
Done.

cliffordp commented on tribe-ext-event-progress-bar.php:
```diff
@@ -0,0 +1,268 @@
+		 * @param $template
+		 */
+		public function progressbar( $file, $name, $template ) {
+
```
Be consistent
```suggestion
```

andrasguseo replied:
Done

cliffordp commented on tribe-ext-event-progress-bar.php:
```diff
@@ -0,0 +1,268 @@
+		 * @return bool
+		 */
+		private function php_version_check() {
+			$php_required_version = '7.0';
```
Add comment why the version you required.

andrasguseo replied:
Bad habit of mine. :)
Done.

cliffordp commented on readme.txt:
```diff
@@ -0,0 +1,40 @@
+Tags: events, calendar
+Requires at least: 5.0
+Tested up to: 5.5.2
+Requires PHP: 7.0
```
Does it really?

andrasguseo replied:
Adjusted and done.

cliffordp reviewed (commented):
_

cliffordp commented on src/event-progress-bar.js:
```diff
@@ -0,0 +1,63 @@
+var x = setInterval( doProgress, 1000 );
+
+function doProgress() {
+
```
Why `var` in some places and `let` in others?

andrasguseo replied:
Simply because I didn't know the difference.
Since then we have re-worked the JavaScript code with Stephen, so this is done.
Done.

andrasguseo commented:
@cliffordp I have addressed or replied to your comments here.

We have been working on the code with Stephen in a separate fork and I addressed your comments in that one, so please check there.
https://github.com/mt-support/tribe-ext-event-progress-bar/pull/2

This fork is deprecated and I'm closing it.
